packages/trader/src/Stores/Modules/Trading/trade-store.ts

```diff
@@ -1389,9 +1412,26 @@ export default class TradeStore extends BaseStore {
         }
 
         if (!this.main_barrier || this.main_barrier?.shade) {
-            this.setMainBarrier(response.echo_req);
+            if (this.is_turbos) {
+                let chart_barrier = (
+                    Number(response.proposal?.contract_details?.barrier) - Number(response.proposal?.spot)
+                ).toFixed(3);
+
+                if (Number(chart_barrier) > 0) {
+                    chart_barrier = `+${chart_barrier}`;
+                }
+                this.setMainBarrier({
+                    ...response.echo_req,
+                    ...{
+                        barrier: String(chart_barrier),
+                    },
+                });
```

Comment:
It's better to wrap this inside an `if` condition to check if the response has proposal to make sure that we handle the error response which will not have the `proposal` object.